testing/scenario_app/lib/src/poppable_screen.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:ui';
+
+import 'platform_echo_mixin.dart';
+import 'scenario.dart';
+
+/// A blank page with a button that pops the page when tapped.
+class PoppableScreenScenario extends Scenario with PlatformEchoMixin {
+  /// Creates the PoppableScreenScenario.
+  ///
+  /// The [window] parameter must not be null.
+  PoppableScreenScenario(Window window)
+      : assert(window != null),
+        super(window);
+
+  // Rect for the pop button. Only defined once onMetricsChanged is called.
+  Rect _buttonRect;
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+    final PictureRecorder recorder = PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+
+    canvas.drawRect(
+      Rect.fromLTWH(0, 0, window.physicalSize.width - 1, window.physicalSize.height - 1),
+      Paint()..color = const Color.fromARGB(255, 255, 255, 255),
+    );
+
+    if (_buttonRect != null) {
+      canvas.drawRect(
+        _buttonRect,
+        Paint()..color = const Color.fromARGB(255, 255, 0, 0),
+      );
+    }
+    final Picture picture = recorder.endRecording();
+
+    builder.pushOffset(0, 0);
+    builder.addPicture(Offset.zero, picture);
+    final Scene scene = builder.build();
+    window.render(scene);
+    scene.dispose();
+  }
+
+  @override
+  void onDrawFrame() {
+    window.scheduleFrame();
```

Comment:
I think you might want this to go to the end of `onMetricsChanged` and `_pop`. Right now this is drawing the same frames over and over again even if nothing has changed.

Reply:
ya, makes sense